File: code/design_pattern/singleton_pattern/SingletonPattern.php
<?php

/**
 * Singleton Design Pattern
 * 
 * @author Anton Vendin
 * @version 1.0
 * 
 */

class Singleton
{

	private static $singletonInstance;

	public static function getInstance() {

		if (static::$singletonInstance === null ) {
			static::$singletonInstance = new Singleton();
		}
		return static::$singletonInstance;
	}
}

$instance = Singleton::getInstance();
print $instance === Singleton::getInstance();

File: code/design_pattern/src/singleton_pattern/singleton_pattern.cs
// Singleton design pattern using properties
class Singleton
{
	private static readonly Singleton _instance = new Singleton();
	public static Singleton Instance { get { return _instance; } }

	private Singleton()
	{
		// Initialize object
	}
}

class Program
{
	static void Main(string[] args)
	{
		var instance = Singleton.Instance;
	}
}